jenkins/Jenkinsfile.macos

```
@@ -96,196 +96,168 @@ pipeline {
       }
     }
 
-    stage("Build Matrix") {
+    stage('Build and Sign') {
 
       environment {
-        ENV = ""
+        AWS_ACCOUNT_ID = '749683154838'
+        KEYCHAIN_PASSPHRASE = credentials('ide-keychain-passphrase')
+        BUILD_AGENT_TEMP = "${env.HOME}/tmp"
+        MACOS_DEVELOPER_CERTIFICATE = credentials('MACOS_DEVELOPER_CERTIFICATE')
+        MACOS_DEVELOPER_CERTIFICATE_KEY = credentials('MACOS_DEVELOPER_CERTIFICATE_KEY')
       }
 
-      matrix {
+      steps {
+        script {
+          ENV = utils.getBuildEnv(!params.DAILY)
+        }
+        script {
+          // Create a keychain to hold the Developer ID certificate for signing
+          sh """
+            if [ ! -f "${BUILD_AGENT_TEMP}/buildagent.keychain" ]; then
+              security create-keychain -p ${KEYCHAIN_PASSPHRASE} ${BUILD_AGENT_TEMP}/buildagent.keychain
+            else
+              echo "Using keychain: ${BUILD_AGENT_TEMP}/buildagent.keychain"
+            fi
+          """
+        }
+        sh 'security default-keychain -s ${BUILD_AGENT_TEMP}/buildagent.keychain'
+        sh 'security unlock-keychain -p ${KEYCHAIN_PASSPHRASE} ${BUILD_AGENT_TEMP}/buildagent.keychain && security set-keychain-settings' // turn off timeout
+        // Import the Developer ID certificate from Jenkins Secrets.
+        sh 'security import \"${MACOS_DEVELOPER_CERTIFICATE}\" -k ${BUILD_AGENT_TEMP}/buildagent.keychain -P \"${MACOS_DEVELOPER_CERTIFICATE_KEY}\" -T /usr/bin/codesign'
+        // build rstudio
+        dir ("package/osx") {
+          withAWS(role: 'build', roleAccount: AWS_ACCOUNT_ID) {
+            sh "${ENV} ./make-package clean --rstudio-target=${FLAVOR}"
+          }
+        }
+      }
+    }
 
-        axes {
-          axis {
-            name 'FLAVOR'
-            values 'Electron'
+    stage('Test') {
+      environment {
+        PATH = "${env.HOME}/opt/bin:${env.PATH}"
+      }
+      steps {
+        dir("package/osx/build/src/cpp") {
+          // attempt to run cpp unit tests
+          // problems with rsession finding openssl, so those tests
+          // are disabled until we solve it (#6890)
+          sh "arch -x86_64 ./rstudio-tests"
+          
+          // electron tests not working in CI environment, under investigation
+          // if(FLAVOR == 'Electron') {
+          //   sh "cd ../../../../../src/node/desktop/ && $HOME/.yarn/bin/yarn && $HOME/.yarn/bin/yarn test"
+          // }
           }
         }
+      }
 
-        when { expression { return FLAVOR == 'Electron' } }
+    stage('Notarize and Upload') {
+      when { expression { return params.PUBLISH } }
+      
+      environment {
+        PATH = "${env.HOME}/opt/bin:${env.PATH}"
+        PACKAGE_FILE = """${sh (
+          script: "basename `ls package/osx/build/RStudio-*.dmg`",
+          returnStdout: true
+        ).trim()}"""
```

I believe the quotes here are superfluous; this should also work fine:

```suggestion
        PACKAGE_FILE = sh(
          script: "basename `ls package/osx/build/RStudio-*.dmg`",
          returnStdout: true
        ).trim()
```